File: Solutions/C#/0019_RemoveNthNodeFromEOL/RemoveNthNode.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        ListNode left = head, right = head;
        for(int i = 0; i < n; i++)
        {
            right = right.next;
        }
        
        if(right == null)
        {
            head = left.next;
            return head;
        }
        
        while(right.next != null)
        {
            left = left.next;
            right = right.next;
        }
        
        left.next = left.next.next;
        return head;
    }
}

File: Solutions/C++/LinkedList/RemoveNthNode.cpp

 struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}ListNode(int x) : val(x), next(nullptr) {}

    ListNode(int x, ListNode *next) : val(x), next(next) {}

};
 class Solution {
 public:
     ListNode* removeNthFromEnd(ListNode* head, int n) {
         ListNode* dummyHead = new ListNode(0, head);

         ListNode *previous = dummyHead, *current = dummyHead;
         while(current != nullptr && n > 0) {
             current = current->next;
             n--;
         }

         if(current == nullptr)
             return dummyHead->next;

         while(true) {
             current = current->next;
             if(current == nullptr)
             {
                 previous->next = previous->next->next;
                 break;
             }

             previous = previous->next;
         }

         return dummyHead->next;
     }
 };